arrows/vtk/CMakeLists.txt

```diff
@@ -25,8 +25,7 @@ kwiver_add_library( kwiver_algo_vtk
 
 target_link_libraries( kwiver_algo_vtk
   PUBLIC               vital
-                       vtkCommonCore
-                       vtkCommonDataModel
+                       ${VTK_LIBRARIES}
```

Comment:
This is a broader question than VTK, but worth bringing up here.  In KWIVER we find all our dependent packages at the root of the project based on the selection of CMake options, but then we don't uses those packages until places like this, down in the arrows.  It is possible (and likely) that different libraries in KWIVER (e.g. kwiver_algo_vtk vs. kwiver_algo_vtk_applets) may need to include different subsets of VTK.  It is not good practice for everything in KWIVER that links to VTK to link to the union of all VTK libraries used across KWIVER.  This is why I was linking to a specific libraries.  Admittedly this doesn't matter just yet, but it will when we add VTK applets very soon.

We can continue to link to specific libraries, as was done here, but my bigger questions is should we also have a separate `find_package` call for each library the uses VTK and then only find the components needed by that library?  Then we could use `${VTK_LIBRARIES}` because it would be local and specific to the list of libraries need for that package.   Or should we have one find_package call, as we do now, with the super set of all things that might be needed and then only include the subset needed for each library?

Reply:
I agree with that comment regarding using different parts of VTK and the extra unnecessary linking. I also don't think it's worth having separate calls to `find_package( VTK` for different uses. I mentioned above with the other comment that we would just need to make the linked `target_link_libraries` call sensitive to the change in naming.

Reply:
What is the disadvantage to having separate calls to `find_package` for each library that wants to use VTK with a different subset of dependencies?  It seems like that would be cleaner and easier to maintain CMake code, since we only need to list the required libraries in one spot for each use case and we would not need this conditional to switch between the new and old library naming conventions.  We could use `${VTK_LIBRARIES}` instead, but `${VTK_LIBRARIES}`  would be redefined each time we use `find_package` for the specific packages needed.

Reply:
That makes sense. I will rearrange the calls later and repush.

Reply:
Has this change been made?

Reply:
Not yet. I will try to get to these fixes this week.